central/complianceoperator/v2/scanconfigurations/service/download.go

```diff
@@ -0,0 +1,138 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	blobDS "github.com/stackrox/rox/central/blob/datastore"
+	snapshotDS "github.com/stackrox/rox/central/complianceoperator/v2/report/datastore"
+	"github.com/stackrox/rox/central/reports/common"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/grpc/authn"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/zip"
+	"google.golang.org/grpc/codes"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+func NewDownloadHandler() http.HandlerFunc {
+	snapshotStore := snapshotDS.Singleton()
+	blobStore := blobDS.Singleton()
+	handler := &downloadHandler{
+		snapshotDataStore: snapshotStore,
+		blobStore:         blobStore,
+	}
+	return handler.handle
+}
+
+type downloadHandler struct {
+	snapshotDataStore snapshotDS.DataStore
+	blobStore         blobDS.Datastore
+}
+
+func (h *downloadHandler) handle(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		httputil.WriteErrorf(w, http.StatusMethodNotAllowed, "Only GET requests are allowed")
+		return
+	}
+
+	id, err := parseReportID(r)
+	if err != nil {
+		httputil.WriteGRPCStyleError(w, codes.InvalidArgument, err)
+		return
+	}
+
+	ctx := r.Context()
+	slimUser := authn.UserFromContext(ctx)
+	if slimUser == nil {
+		httputil.WriteGRPCStyleError(w, codes.PermissionDenied, errors.New("Could not determine user identity from provided context"))
+		return
+	}
+
+	snapshot, found, err := h.snapshotDataStore.GetSnapshot(ctx, id)
+	if err != nil {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.Errorf("Error retrieving snapshot with the given id %s", id))
+		return
+	}
+	if !found {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.Errorf("Unable to find the snapshot with the given id %s", id))
+		return
+	}
+
+	if slimUser.GetId() != snapshot.GetUser().GetId() {
+		httputil.WriteGRPCStyleError(w, codes.PermissionDenied, errors.Errorf("Report %s cannot be downloaded by the user %s", id, slimUser.GetId()))
+		return
+	}
```

Comment:
This allows a user who has no permissions to check whether the snapshot exists or not - is this a security issue?

Reply:
This a very valid concern. This is how it's done in VM reporting, since we want to have as much feature parity as possible, I'd either keep it like this or change it in both places.

Reply:
Okay to keep, okay to change in both places in a followup (as discussed in the standup Today).